server_test.go

```diff
@@ -3350,6 +3350,110 @@ func TestMaxBodySizePerRequest(t *testing.T) {
 	}
 }
 
+func TestStreamRequestBody(t *testing.T) {
+	t.Parallel()
+
+	part1 := strings.Repeat("1", 1<<10)
+	part2 := strings.Repeat("2", 1<<20-1<<10)
+
+	reqBody := part1 + part2
+	contentLength := len(reqBody)
+
+	s := &Server{
+		Handler: func(ctx *RequestCtx) {
+			checkReader(t, ctx.RequestBodyStream(), reqBody)
+		},
+		ReadTimeout:       time.Second * 5,
+		StreamRequestBody: true,
+	}
+
+	rw := &readWriter{}
+	rw.r.WriteString(fmt.Sprintf("POST /foo2 HTTP/1.1\r\nHost: aaa.com\r\nContent-Length: %d\r\nContent-Type: aa\r\n\r\n%s", contentLength, part1))
+
```

Comment:
What I would like to see if for you to do a read on the request body here and see if it matches `part1`. Then `rw.r.WriteString(part2)` and then read the rest of the request body and see if it matches `part2`. That way you test if you can already read the partial body before the full body it send. Right now you're not testing that.